pkgs/pkgs-lib/formats/libconfig/test/comprehensive/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib, formats, stdenvNoCC, writeText, ... }:
+let
+  libconfig = formats.libconfig { };
+
+  include_expr = {
+    val = 1;
+  };
+
+  include_file = writeText "libconfig-test-include" ''
+    val=1;
+  '';
```

Comment:
This is a regular input-addressed derivation, which means that the output path will change whenever anything about the implementation changes, ie some updates to Nixpkgs.

You could change it to `builtins.toFile`, but that would turn it from a package attrset to a simple string with context, which I don't think is what you want to test here? Perhaps a better alternative is to make this a fixed output derivation, such as what can be done with a fetcher. That will make for unchanging output paths, as suggested by "fixed output".

Reply:
The intent of the test is to verify that the output of having files in `_includes` (which are most likely to be nix store paths) are accepted as valid config by the upstream library.

I could make it a FOD, but I don't really see how it would help. What kind of changes to the lowlevel implementations in nixpkgs could change the hash of a static text file, without also affecting the hashes of FODs? Extra meta attributes in `writeFile`?

Reply:
> I could make it a FOD, but I don't really see how it would help. What kind of changes to the lowlevel implementations in nixpkgs could change the hash of a static text file, without also affecting the hashes of FODs? Extra meta attributes in `writeFile`?

Apparently quite a lot:
![git blame of the writeTextFile definition](https://github.com/NixOS/nixpkgs/assets/25263210/c8885b47-fb0c-4ce9-8469-067cb13f238c)

And this did just happen: https://github.com/NixOS/nixpkgs/pull/246115#issuecomment-1763525016

Reply:
> What kind of changes to the lowlevel implementations in nixpkgs could change the hash of a static text file

None. The _contents_ of the _output_ of `writeTextFile` is very very stable, which is why you want a content-addressed output instead of an input addressed output. By making it a fixed-output derivation that's exactly what you get. It shields you from the many obscure changes that happen to the derivation closure.

How output path hashes are derived would be a good topic for an explanation, but what we have now is reference docs, https://nixos.org/manual/nix/stable/language/advanced-attributes.html?highlight=outputHash#adv-attr-outputHash

Reply:
Thank you. This will be reading material for later!

I've updated it to a FOD, so this should be fine now.